File: leetcode/1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        array<int,26> fq1={0};
        array<int,26> fq2={0};
        if(word1.length()!=word2.length()){
            return false;
        }
        for(int i=0;i<word1.length();i++){
            int ind=word1[i]-'a';
            fq1[ind]++;
        }
        for(int j=0;j<word2.length();j++){
            int ind=word2[j]-'a';
            fq2[ind]++;
        }
        for(int i=0;i<26;i++){
            if((fq1[i]==0 && fq2[i]!=0)||(fq1[i]!=0 && fq2[i]==0)){
                return false;
            }
        }
        sort(fq1.begin(),fq1.end());
        sort(fq2.begin(),fq2.end());
        for(int i=0;i<26;i++){
            if(fq1[i]!=fq2[i]){
                return false;
            }
        }
        return true;
    }
};

File: leetcode/1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.py
class Solution(object):
    def closeStrings(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: bool
        """
        fq1 = {}
        fq2 = {}
        for w in word1:
            if w in fq1.keys():
                fq1[w] += 1
            else:
                fq1[w] = 1
        
        for w in word2:
            if w in fq2.keys():
                fq2[w] += 1
            else:
                fq2[w] = 1

        return set(fq1.keys()) == set(fq2.keys()) and sorted(fq1.values())== sorted(fq2.values())